sources/osgShader/node/billboard.glsl

```diff
@@ -0,0 +1,16 @@
+vec3 getScale( const in mat4 matrix ) {
+      // Only working with positive scales.
+      float xs = matrix[0][0] * matrix[0][1] * matrix[0][2] * matrix[0][3] < 0. ? -1. : 1.;
+      float ys = matrix[1][0] * matrix[1][1] * matrix[1][2] * matrix[1][3] < 0. ? -1. : 1.;
+      float zs = matrix[2][0] * matrix[2][1] * matrix[2][2] * matrix[2][3] < 0. ? -1. : 1.;
+      vec3 scale;
+      scale.x = xs * sqrt( matrix[0][0] * matrix[0][0] + matrix[0][1] * matrix[0][1] + matrix[0][2] * matrix[0][2]);
+      scale.y = ys * sqrt( matrix[1][0] * matrix[1][0] + matrix[1][1] * matrix[1][1] + matrix[1][2] * matrix[1][2]);
+      scale.z = zs * sqrt( matrix[2][0] * matrix[2][0] + matrix[2][1] * matrix[2][1] + matrix[2][2] * matrix[2][2]);
+      return scale;
+}
+
+vec4 billboard( const in vec3 vertex, const in mat4 modelViewMatrix, const in mat4 projectionMatrix ) {
+      vec3 scale = getScale( modelViewMatrix );
+      return projectionMatrix * ( vec4( scale.x* vertex.x , scale.y * vertex.y, scale.z * vertex.z, 1.0 ) + vec4( modelViewMatrix[ 3 ].xyz, 0.0 ) );
```

Comment:
Seems very costly method to me, first time I see that much complexity in billboard shader.
Not even sure to get what it does.
perhaps 
_projectionMatrix_ **+** _modelViewMatrix_ 
should be 
_projectionMatrix_ **x** _modelViewMatrix_ 

I recall other faster methods for billboard but perhaps doesn't fit your specifications/needs  ?
Perhaps add a comment on why this one needed vs all the other ones like the ones in [this list](http://www.opengl-tutorial.org/fr/intermediate-tutorials/billboards-particles/billboards/)